### PR TITLE
Add latest participation listing on contact detail views

### LINKED ISSUE #2108: List all dossiers of a Contact
We want to list all dossiers where a `Contact` participates. Performance may be an issue with `IntId` lookup, investigate.

### PR DESCRIPTION
**Pull request by phgross**
Closes #2108 

bildschirmfoto 2016-08-30 um 14 07 21

To avoid performance issues, by fetching all dossiers over the IntId utility, the listing is implemented asynchronous and limited to 50 items. Listing all participations ist possible with a "show all" link.

@deiferni

### CODE REVIEW

**Comment by deiferni** [comment]
I would move the "Show all Participations" button/link below the already listed participations

File: opengever/contact/browser/participations.py [hunk 1]
```diff
@@ -1,9 +1,57 @@
...
+
+    def get_slice_size(self):
+        return api.portal.get_registry_record(
+            'batch_size', interface=ITabbedView)
+
+    def list(self):
+        """Returns a json dict with the following structure.
+
+        `participations`: a list of json representation of the current
+         contexts participations. It sliced to the default tabbedview batch
```
Comment (deiferni) [inline opengever/contact/browser/participations.py]: whitespace is off ✂️

File: opengever/contact/browser/participations.py [hunk 2]
```diff
@@ -1,9 +1,57 @@
...
+
+        data = {}
+        query = Participation.query.filter_by(contact=self.context.model)
+        query = query.order_by(desc(Participation.participation_id))
+        total = query.count()
+
+        if self.request.get('show_all') != 'true':
+            query = query.limit(self.get_slice_size())
+
+        data['participations'] = self._serialize(query.all())
```
Comment (deiferni) [inline opengever/contact/browser/participations.py]: the query is iterable, i think you could say `self._serialize(query)` this won't load all results at once

**Comment by deiferni** [comment]
Some details, rest looks 👍

**Comment by deiferni** [comment]
oh and another thought: this won't list the participations for a persons `OrgRole`, so we might need to update after both still open PRs have been merged

**Comment by phgross** [comment]
@deiferni updated. 

I would handle the OrgRole participations in a separate PR.

**Pull request merged**